protocol_test.go

```diff
@@ -29,6 +29,7 @@ func TestCanonicalizeContentType(t *testing.T) {
 	}{
 		{name: "charset param should be treated as lowercase", arg: "application/json; charset=UTF-8", want: "application/json; charset=utf-8"},
 		{name: "non charset param should not be changed", arg: "multipart/form-data; boundary=fooBar", want: "multipart/form-data; boundary=fooBar"},
+		{name: "no parameters should not have base normalized", arg: "APPLICATION/json;  ", want: "application/json"},
```

Comment:
Either the name or the assertion seems wrong to me. Should the description be "no parameters _should_ have base normalized"?

Reply:
wups, lemme fix